core/src/main/java/com/nisovin/magicspells/spells/targeted/StructureSpell.java

```diff
@@ -0,0 +1,87 @@
+package com.nisovin.magicspells.spells.targeted;
+
+import org.bukkit.Bukkit;
+import org.bukkit.Location;
+import org.bukkit.util.Vector;
+import org.bukkit.NamespacedKey;
+import org.bukkit.structure.Structure;
+import org.bukkit.block.structure.Mirror;
+import org.bukkit.block.structure.StructureRotation;
+
+import com.nisovin.magicspells.util.SpellData;
+import com.nisovin.magicspells.util.CastResult;
+import com.nisovin.magicspells.util.TargetInfo;
+import com.nisovin.magicspells.util.MagicConfig;
+import com.nisovin.magicspells.spells.TargetedSpell;
+import com.nisovin.magicspells.util.config.ConfigData;
+import com.nisovin.magicspells.spells.TargetedLocationSpell;
+import com.nisovin.magicspells.events.SpellTargetLocationEvent;
+
+public class StructureSpell extends TargetedSpell implements TargetedLocationSpell {
+	private final ConfigData<String> structureKey;
+	private final ConfigData<StructureRotation> rotation;
+	private final ConfigData<Mirror> mirror;
+	private final ConfigData<Integer> palette;
+	private final ConfigData<Float> integrity;
+	private final ConfigData<Boolean> includeEntities;
+
+	private final ConfigData<Vector> relativeOffset;
+
+	private final ConfigData<Boolean> placeAtCaster;
+	public StructureSpell(MagicConfig config, String spellName) {
+		super(config, spellName);
+
+		structureKey = getConfigDataString("structure-key", "");
+
+		rotation = getConfigDataEnum("rotation", StructureRotation.class, StructureRotation.NONE);
+		mirror = getConfigDataEnum("mirror", Mirror.class, Mirror.NONE);
+		palette = getConfigDataInt("palette", 0);
+		integrity = getConfigDataFloat("integrity", 1);
+		includeEntities = getConfigDataBoolean("include-entities", false);
+
+		relativeOffset = getConfigDataVector("relative-offset", new Vector());
+
+		placeAtCaster = getConfigDataBoolean("place-at-caster", false);
```

Comment:
Name the option `point-blank` for consistency.

Reply:
Changed in newest commit